modules/drivers/databricks/src/metabase/driver/databricks.clj

```diff
@@ -201,29 +201,38 @@
     (str/replace-first additional-options #"^(?!;)" ";")))
 
 (defmethod sql-jdbc.conn/connection-details->spec :databricks
-  [_driver {:keys [catalog host http-path log-level token additional-options] :as _details}]
+  [_driver {:keys [catalog host http-path use-m2m token client-id oauth-secret log-level additional-options] :as _details}]
   (assert (string? (not-empty catalog)) "Catalog is mandatory.")
-  (merge
-   {:classname        "com.databricks.client.jdbc.Driver"
-    :subprotocol      "databricks"
-    ;; Reading through the changelog revealed `EnableArrow=0` solves multiple problems. Including the exception logged
-    ;; during first `can-connect?` call. Ref:
-    ;; https://databricks-bi-artifacts.s3.us-east-2.amazonaws.com/simbaspark-drivers/jdbc/2.6.40/docs/release-notes.txt
-    :subname          (str "//" host ":443/;EnableArrow=0"
-                           ";ConnCatalog=" (codec/url-encode catalog)
-                           (preprocess-additional-options additional-options))
-    :transportMode  "http"
-    :ssl            1
-    :AuthMech       3
-    :HttpPath       http-path
-    :uid            "token"
-    :pwd            token
-    :UserAgentEntry (format "Metabase/%s" (:tag config/mb-version-info))
-    :UseNativeQuery 1}
-   ;; Following is used just for tests. See the [[metabase.driver.sql-jdbc.connection-test/perturb-db-details]]
-   ;; and test that is using the function.
-   (when log-level
-     {:LogLevel log-level})))
+  (let [base-spec
+        {
+          :classname      "com.databricks.client.jdbc.Driver"
+          :subprotocol    "databricks"
+          ;; Reading through the changelog revealed `EnableArrow=0` solves multiple problems. Including the exception logged
+          ;; during first `can-connect?` call. Ref:
+          ;; https://databricks-bi-artifacts.s3.us-east-2.amazonaws.com/simbaspark-drivers/jdbc/2.6.40/docs/release-notes.txt
+          :subname        (str "//" host ":443/;EnableArrow=0"
+                              ";ConnCatalog=" (codec/url-encode catalog)
+                              (preprocess-additional-options additional-options))
+          :transportMode  "http"
+          :ssl            1
+          :HttpPath       http-path
+          :UserAgentEntry (format "Metabase/%s" (:tag config/mb-version-info))
+          :UseNativeQuery 1}
+          ;; Following is used just for tests. See the [[metabase.driver.sql-jdbc.connection-test/perturb-db-details]]
+          ;; and test that is using the function.
+        base-spec (if log-level (assoc base-spec :LogLevel log-level) base-spec)]
+    (if use-m2m
+      ;; M2M OAuth
+      (assoc base-spec
+             :AuthMech 11
+             :Auth_Flow 1
+             :OAuth2ClientId client-id
+             :OAuth2Secret oauth-secret)
+      ;; PAT authentication
+      (assoc base-spec
+             :AuthMech 3
+             :uid "token"
+             :pwd token))))
```

Comment:
nit: (1) whitespace, (2) I think rewriting this function in a form as follows would make it cleaner.
```
(merge 
  {base}
  (when log-level
    {:LogLevel log-level})
  (if use-m2m
    {:AuthMech 11,,,}
    {:AuthMech 3,,,}))
```

Reply:
I've tried to implement your feedback. Please take a look and resolve if OK